### PR TITLE
Feature/nuget compare

### PR DESCRIPTION
**Pull request by gponomarev-lykke**

### CODE REVIEW

File: src/Lykke.NuGetReferencesScanner.Domain/IReferencesScanner.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using System.Threading.Tasks;
+
+namespace Lykke.NuGetReferencesScanner.Domain
+{
+    public interface IReferencesScanner
+    {
+        Task<ScanResult> GetScanResult();
```
Comment (starkmsu) [inline src/Lykke.NuGetReferencesScanner.Domain/IReferencesScanner.cs:7]: Please rename to GetScanResultAsync

**Comment by gponomarev-lykke** [comment]
I changed configuration for both Lykke.NuGetReferencesScanner and Lykke.NuGetReferencesScanner.App, an example can be found in readme.md in the Lykke.NuGetReferencesScanner.App project.

**Comment by gponomarev-lykke** [comment]
The default behaviour of Lykke.NuGetReferencesScanner has been changed. It now scans all found packages. The previous behaviour with regex-based filters can be achieved by setting "mode" config parameter to "include".

**Comment by gponomarev-lykke** [comment]
I could not run GithubScanner with OAuth authorization. Based on this info https://github.com/octokit/octokit.net/blob/main/docs/oauth-flow.md a more complicated flow is required to generate the access token. I tested the app with login / password. Because of this I skipped Github configuration in readme.

File: src/Lykke.NuGetReferencesScanner.Domain/PackageWhitelist.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+namespace Lykke.NuGetReferencesScanner.Domain
+{
+    public interface IPackageWhitelist
+    {
+        bool ShouldSkip(string repo);
+    }
+
+    public class PackageWhitelist : IPackageWhitelist
+    {
+        private List<string> _whitelist;
```
Comment (tarurar) [inline src/Lykke.NuGetReferencesScanner.Domain/PackageWhitelist.cs:14]: Can be made readonly

**Pull request merged**